pkg/datamover/datamover.go

```diff
@@ -122,3 +123,174 @@ func CheckIfVolumeSnapshotBackupsAreComplete(ctx context.Context, volumesnapshot
 	}
 	return eg.Wait()
 }
+
+func GetRestoreClient() (kbclient.Client, error) {
+	client, err := kbclient.New(config.GetConfigOrDie(), kbclient.Options{})
+	if err != nil {
+		return nil, err
+	}
+	err = velerov1api.AddToScheme(client.Scheme())
+	if err != nil {
+		return nil, err
+	}
+
+	return client, err
+}
+
+func CleanupRestoreVSRs(currentRestore *velerov1api.Restore, log logrus.FieldLogger) error {
+	if err := DeleteVSRsIfComplete(currentRestore.Name, log); err != nil {
+		return err
+	}
+
+	// Check other completed Restores for VSRs that can be cleaned up
+	restoreList := velerov1api.RestoreList{}
+	client, err := GetRestoreClient()
+	if err != nil {
+		return err
+	}
+
+	err = client.List(context.TODO(), &restoreList, &kbclient.ListOptions{})
+	if err != nil {
+		log.Errorf(err.Error())
+		return err
+	}
+
+	for _, restore := range restoreList.Items {
```

Comment:
We are calling the `CleanupRestoreVSRs` function for the restores with completed phase, IMO we should only cleanup VSRs for this current restore and not touch any other VSRs.

Reply:
@shubham-pampattiwar Hmm. I guess that depends on whether we want to clean up old non-cleaned restores or not. If we're only ever concerned with removing VSRs for the current restore, then this should be simpler.

Reply:
clean up only the vsr's associated w/ the restore in process.

Reply:
I put this back to only removing VSRs for the current restore.